iceoryx_posh/test/moduletests/test_popo_untyped_publisher.cpp

```diff
@@ -41,14 +41,97 @@ class UntypedPublisherTest : public Test
   protected:
     ChunkMock<uint64_t> chunkMock;
     TestUntypedPublisher sut{{"", "", ""}};
+    MockPublisherPortUser& portMock{sut.mockPort()};
 };
 
-TEST_F(UntypedPublisherTest, PublishesVoidPointerViaUnderlyingPort)
+TEST_F(UntypedPublisherTest, LoansChunkWithRequestedSize)
+{
+    constexpr uint32_t ALLOCATION_SIZE = 7;
+    EXPECT_CALL(portMock, tryAllocateChunk(ALLOCATION_SIZE))
+        .WillOnce(Return(ByMove(iox::cxx::success<iox::mepoo::ChunkHeader*>(chunkMock.chunkHeader()))));
+    // ===== Test ===== //
+    auto result = sut.loan(ALLOCATION_SIZE);
+    // ===== Verify ===== //
+    EXPECT_FALSE(result.has_error());
+    // ===== Cleanup ===== //
+}
+
+TEST_F(UntypedPublisherTest, LoanFailsIfPortCannotSatisfyAllocationRequest)
+{
+    constexpr uint32_t ALLOCATION_SIZE = 17;
+    EXPECT_CALL(portMock, tryAllocateChunk(ALLOCATION_SIZE))
+        .WillOnce(Return(
+            ByMove(iox::cxx::error<iox::popo::AllocationError>(iox::popo::AllocationError::RUNNING_OUT_OF_CHUNKS))));
+    // ===== Test ===== //
+    auto result = sut.loan(ALLOCATION_SIZE);
+    // ===== Verify ===== //
+    ASSERT_TRUE(result.has_error());
+    EXPECT_EQ(iox::popo::AllocationError::RUNNING_OUT_OF_CHUNKS, result.get_error());
+    // ===== Cleanup ===== //
+}
+
+TEST_F(UntypedPublisherTest, LoanPreviousChunkSucceeds)
+{
+    EXPECT_CALL(portMock, tryGetPreviousChunk())
+        .WillOnce(Return(ByMove(iox::cxx::optional<iox::mepoo::ChunkHeader*>(chunkMock.chunkHeader()))));
+    // ===== Test ===== //
+    auto result = sut.loanPreviousChunk();
+    // ===== Verify ===== //
+    EXPECT_TRUE(result.has_value());
+    // ===== Cleanup ===== //
+}
+
+TEST_F(UntypedPublisherTest, LoanPreviousChunkFails)
+{
+    EXPECT_CALL(portMock, tryGetPreviousChunk()).WillOnce(Return(ByMove(iox::cxx::nullopt)));
+    // ===== Test ===== //
+    auto result = sut.loanPreviousChunk();
+    // ===== Verify ===== //
+    EXPECT_FALSE(result.has_value());
+    // ===== Cleanup ===== //
+}
+
+TEST_F(UntypedPublisherTest, PublishesPayloadViaUnderlyingPort)
 {
     // ===== Setup ===== //
-    EXPECT_CALL(sut.m_port, sendChunk).Times(1); // m_port is mocked.
+    EXPECT_CALL(portMock, sendChunk).Times(1);
     // ===== Test ===== //
     sut.publish(chunkMock.chunkHeader()->payload());
     // ===== Verify ===== //
     // ===== Cleanup ===== //
 }
+
+// test whether the BasePublisher methods are called
```

Comment:
@elBoberido 
I would prefer typed tests but this is a bit cumbersome here, and besides, where should they live (most reasonable thing would be the base publisher tests, but it is not ideal).

In general I recommend testing the functionality for the derived classes as well, since you may expect them to be the same and yet it might change due to e.g. overriding a virtual private method which alters the behavior of any some public function it is used in. There are other scenarios as well.

Here it is just simple delegation and no such thing as described happens, but to be safe it is not wrong to have them. Just more code and test execution time.